src/opnsense/scripts/unbound-dnsbl/lib/dnsbl.py

```diff
@@ -64,40 +63,42 @@ def _update_dnsbl(self):
+        last_state = self.dnsbl is not None
+
+        if not self._dnsbl_exists():
+            self.dnsbl = None
+        else:
             fstat = os.stat(self.dnsbl_path).st_mtime
             if fstat != self.dnsbl_mtime_cache:
                 self.dnsbl_mtime_cache = fstat
                 log_info("dnsbl_module: updating blocklist.")
-                self._load_dnsbl()
-
-    def _load_dnsbl(self):
-        with open(self.dnsbl_path, 'r') as f:
-            try:
-                self.dnsbl = json.load(f)
-                if self._context and type(self.dnsbl.get('config')) is dict:
-                    if not self.dnsbl['config'].get('general'):
-                        # old format, needs blocklist reload
-                        raise ValueError("incompatible blocklist")
-                    self._context.set_config(self.dnsbl['config'])
-                log_info('dnsbl_module: blocklist loaded. length is %d' % len(self.dnsbl['data']))
-                with open(self.size_file, 'w') as sfile:
-                    sfile.write(str(len(self.dnsbl['data'])))
-            except (json.decoder.JSONDecodeError, KeyError, ValueError) as e:
-                if not self.dnsbl or isinstance(e, ValueError):
-                    log_err("dnsbl_module: unable to parse blocklist file: %s. Please re-apply the blocklist settings." % e)
-                    self.dnsbl_available = False
-                    open(self.warn_file, "a").close()
-                    return
-                else:
-                    log_err("dnsbl_module: error parsing blocklist: %s, reusing last known list" % e)
+                with open(self.dnsbl_path, 'r') as f:
+                    try:
+                        self.dnsbl = json.load(f)
+                        if self._context and type(self.dnsbl.get('config')) is dict:
+                            if not self.dnsbl['config'].get('general'):
+                                # old format, needs blocklist reload
+                                self.dnsbl = None
+                                raise ValueError("incompatible blocklist")
+                            self._context.set_config(self.dnsbl['config'])
+                        log_info('dnsbl_module: blocklist loaded. length is %d' % len(self.dnsbl['data']))
+                    except (json.decoder.JSONDecodeError, KeyError, ValueError) as e:
+                        if not self.dnsbl or isinstance(e, ValueError):
+                            log_err("dnsbl_module: unable to parse blocklist file: %s. Please re-apply the blocklist settings." % e)
+                            open(self.warn_file, "a").close()
+                            return
+                        else:
+                            log_err("dnsbl_module: error parsing blocklist: %s, reusing last known state" % e)
 
         if os.path.exists(self.warn_file):
             os.remove(self.warn_file)
 
-        self.dnsbl_available = True
+        if last_state != self.dnsbl is not None:
```

Comment:
```suggestion
        if last_state != (self.dnsbl is not None):
```

It might be correct as it was, I'm just doubting here.

Reply:
good catch, as it was it would be equivalent to `if last_state != self.dnsbl and self.dnsbl is not None`, the original `last_state` should be changed as well